src/Executor/Setup.cs

```diff
@@ -0,0 +1,290 @@
+            foreach (KeyValuePair<string, string> parameters in settingParameters)
+            {
+                if (!string.IsNullOrEmpty(parameters.Value))
+                {
+                    commandSetting += string.Format(" {0}", FormatSetting(parameters.Key, parameters.Value, FindSettingType(parameters.Key), toolName));
+                }
+            }
+            return commandSetting;
+        }
+
+        private bool BuildRequiredValueSettingsForCommand(Dictionary<string, string> requiredSettings, Dictionary<string, string> commandValues)
+        {
+            foreach (KeyValuePair<string, string> reqSetting in requiredSettings)
+            {
+                string value = string.IsNullOrEmpty(reqSetting.Value) || reqSetting.Value.Equals("default") ? FindSettingValue(reqSetting.Key) : reqSetting.Value;
+                if (value != null && (string.IsNullOrEmpty(commandValues[reqSetting.Key]) || reqSetting.Key.Equals("Project")))
+                {
+                    commandValues[reqSetting.Key] = string.IsNullOrEmpty(value) ? "True" : value;
+                }
+                else
+                {
+                    if (!string.IsNullOrEmpty(value) && !value.Equals(commandValues[reqSetting.Key]))
+                    {
+                        Console.WriteLine("Error: The value for setting {0} can't be overwriten.", reqSetting.Key);
+                        return false;
+                    }
+                }
+            }
+            return true;
+        }
+
+        private bool BuildOptionalValueSettingsForCommand(Dictionary<string, string> optionalSettings, Dictionary<string, string> commandValues)
+        {
+            foreach (KeyValuePair<string, string> optSetting in optionalSettings)
+            {
+                if (string.IsNullOrEmpty(commandValues[optSetting.Key]))
+                {
+                    string value = string.IsNullOrEmpty(optSetting.Value) || optSetting.Value.Equals("default") ? FindSettingValue(optSetting.Key) : optSetting.Value;
+                    if (value != null)
+                    {
+                        commandValues[optSetting.Key] = value;
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                }
+            }
+            return true;
+        }
+
+        private bool ValidExtraArgumentsForCommand(string extraArguments, Dictionary<string, string> commandValues)
+        {
+            int colonPosition;
+            int equalPosition;
+            string tempParam;
+
+            string[] extraA = extraArguments.Split(' ');
+            foreach(string param in extraA)
+            {
+                colonPosition = 0;
+                equalPosition = param.Length;
+                tempParam = string.Empty;
+
+                colonPosition = param.IndexOf(":");
+                equalPosition = param.IndexOf("=");
+                if(colonPosition!=0)
```

Comment:
True, but do we have parameters like that? I looked at the documentation for msbuild and cmake to make sure I'm covering those scenarios.
I found that this was not working for `foo=bar` that is a possible parameter for cmake so I fixed it.